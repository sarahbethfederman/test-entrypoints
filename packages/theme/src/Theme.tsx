import * as React from 'react';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;
  }

  input::-ms-clear {
    display: none;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors['shade.700']};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>
        {children}
        <GlobalStyle />
      </Root>
    </ThemeProvider>
  );
}
