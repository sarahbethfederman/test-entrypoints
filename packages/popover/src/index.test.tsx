/* tslint:disable */
import * as React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Theme from '@lendi-ui/theme';
import Popover from './index';
import { Position } from './index.style';

import '@testing-library/jest-dom/extend-expect';

const popoverContent = 'This is the content!';
const buttonText = 'Click me';
const PopoverComponent = () => (
  <Theme>
    <Popover position="top">
      <Popover.Target>
        <button>{buttonText}</button>
      </Popover.Target>
      <Popover.Content>{popoverContent}</Popover.Content>
    </Popover>
  </Theme>
);

// The tests throw a bunch of errors if they can't see these methods
// these were taken from the library's tests here:
// https://github.com/mohsinulhaq/react-popper-tooltip/blob/master/tests/TooltipTrigger.spec.tsx
// @ts-ignore
window.MutationObserver = class {
  public disconnect() {}
  public observe() {}
};

// @ts-ignore
if (global.document) {
  document.createRange = () => ({
    setStart: () => {},
    setEnd: () => {},
    // @ts-ignore
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
  });
}

jest.useFakeTimers();
afterEach(cleanup);

describe('Popover', () => {
  /**
   * So this is an issue described in this thread:
   * https://github.com/FezVrasta/popper.js/issues/478
   * The tests will pass regardless, but since it uses JSDOM
   * and not the real DOM API, the creator has recognised
   * that these warnings will appear, swallowing them here
   * for convenience.
   */
  let error;
  beforeEach(() => {
    error = console.error;
    console.error = () => {};
  });

  afterEach(() => {
    console.error = error;
  });

  describe('intial render', () => {
    let getByText: any;
    let queryByText: any;

    beforeEach(() => {
      ({ getByText, queryByText } = render(<PopoverComponent />));
    });

    it('should render the target button', () => {
      expect(getByText(buttonText)).toBeInTheDocument();
    });

    it('should not initially render the textContent', () => {
      expect(queryByText(popoverContent)).toBeNull();
    });
  });

  describe('on click behaviour', () => {
    let getByText: any;
    let queryByText: any;

    beforeEach(() => {
      ({ getByText, queryByText } = render(<PopoverComponent />));
      fireEvent.click(getByText(buttonText));
      jest.runAllTimers();
    });

    it('should display the popoverContent when the component is clicked', () => {
      expect(getByText(popoverContent)).toBeInTheDocument();
    });

    it('should hide the text when the target is clicked again', () => {
      fireEvent.click(getByText(buttonText));
      jest.runAllTimers();
      expect(queryByText(popoverContent)).toBeNull();
    });

    it('should hide thext when anything outside the component is clicked', () => {
      fireEvent.click(document.body);
      jest.runAllTimers();
      expect(queryByText(popoverContent)).toBeNull();
    });
  });

  describe('position based tests', () => {
    const positions = [
      'top',
      'top-end',
      'top-start',
      'bottom',
      'bottom-end',
      'bottom-start',
      'left',
      'left-end',
      'left-start',
      'right',
      'right-end',
      'right-start',
    ];

    positions.forEach((pos: Position) => {
      it(`should be set correctly in the ${pos} position`, () => {
        const { getByText } = render(
          <Theme>
            <Popover position={pos}>
              <Popover.Target>
                <button>{buttonText}</button>
              </Popover.Target>
              <Popover.Content>{popoverContent}</Popover.Content>
            </Popover>
          </Theme>
        );
        fireEvent.click(getByText(buttonText));
        jest.runAllTimers();

        const parentContainer = getByText(popoverContent).parentElement;
        expect(parentContainer).toMatchSnapshot();

        const arrowContainer = parentContainer.firstChild.firstChild;
        expect(arrowContainer).toHaveAttribute('data-placement', pos);
      });
    });
  });

  describe('native props & standard HTML attributes', () => {
    let target: any;
    beforeEach(() => {
      const { getByText, getByTestId } = render(
        <Theme>
          <Popover
            position="top"
            aria-label="testLabel"
            aria-describedby="info"
            id="testId"
            title="testTitle"
            data-testid="popover"
          >
            <Popover.Target>
              <button>{buttonText}</button>
            </Popover.Target>
            <Popover.Content>{popoverContent}</Popover.Content>
          </Popover>
        </Theme>
      );
      fireEvent.click(getByText(buttonText));
      jest.runAllTimers();
      target = getByTestId('popover');
    });

    it('should mount with aria attributes', () => {
      expect(target).toHaveAttribute('aria-label', 'testLabel');
      expect(target).toHaveAttribute('aria-describedby', 'info');
    });

    it('should mount with native props like id, title', () => {
      expect(target).toHaveAttribute('id', 'testId');
      expect(target).toHaveAttribute('title', 'testTitle');
    });
  });
});
/* tslint:enable */
