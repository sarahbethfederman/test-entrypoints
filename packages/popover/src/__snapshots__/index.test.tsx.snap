// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover position based tests should be set correctly in the bottom position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="bottom"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the bottom-end position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="bottom-end"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the bottom-start position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="bottom-start"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the left position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="left"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the left-end position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="left-end"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the left-start position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="left-start"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the right position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="right"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the right-end position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="right-end"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the right-start position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="right-start"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the top position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="top"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the top-end position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="top-end"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;

exports[`Popover position based tests should be set correctly in the top-start position 1`] = `
.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(10,72,87,1);
  color: rgba(255,255,255,1);
  text-align: left;
  border-radius: 5px;
  padding: var(--lendi-ui-size);
}

.c0 .tooltip-arrow {
  width: 0;
  height: 0;
  border-style: solid;
  position: absolute;
  margin: 5px;
}

.c0 .tooltip-arrow[data-placement*='bottom'] {
  border-width: 0 5px 5px 5px;
  border-color: transparent transparent rgba(10,72,87,1) transparent;
  top: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='top'] {
  border-width: 5px 5px 0 5px;
  border-color: rgba(10,72,87,1) transparent transparent transparent;
  bottom: -5px;
  left: calc(50% - 5px);
  margin-top: 0;
  margin-bottom: 0;
}

.c0 .tooltip-arrow[data-placement*='right'] {
  border-width: 5px 5px 5px 0;
  border-color: transparent rgba(10,72,87,1) transparent transparent;
  left: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

.c0 .tooltip-arrow[data-placement*='left'] {
  border-width: 5px 0 5px 5px;
  border-color: transparent transparent transparent rgba(10,72,87,1);
  right: -5px;
  top: calc(50% - 5px);
  margin-left: 0;
  margin-right: 0;
}

<div
  class="tooltip-container"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
>
  <div
    class="c0"
  >
    <div
      class="tooltip-arrow"
      data-placement="top-start"
    />
  </div>
  <div
    class="c1"
  >
    This is the content!
  </div>
</div>
`;
